native-sql-engine/core/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala

```diff
@@ -338,12 +338,17 @@ case class ColumnarPostOverrides() extends Rule[SparkPlan] {
 
   def replaceWithColumnarPlan(plan: SparkPlan): SparkPlan = plan match {
     case plan: RowToColumnarExec =>
+      val child = replaceWithColumnarPlan(plan.child)
       if (columnarConf.enableArrowRowToColumnar) {
-        val child = replaceWithColumnarPlan(plan.child)
         logDebug(s"ColumnarPostOverrides ArrowRowToColumnarExec(${child.getClass})")
-        new ArrowRowToColumnarExec(child)
+        try {
+          new ArrowRowToColumnarExec(child)
+        } catch {
+          case _: Throwable =>
+            logInfo("ArrowRowToColumnar: Falling back to RowToColumnar...")
+            RowToArrowColumnarExec(child)
```

Comment:
@haojinIntel hi, I added ArrayType to buildCheck. But the fallback logics may be still needed to prevent throwing exception on other unsupported cases. Please take a review for that, thanks.